src/tools/plan-generator.ts

```diff
@@ -62,7 +62,7 @@ export class PlanGenerator {
 
       const planArgs = [ktUtils.listOf(particles)];
       if (arcId) {
-        planArgs.push(quote(arcId));
+        planArgs.push(`${quote(arcId)}.toArcId()`);
```

Comment:
Make sure to update the Goldens: 
- run `bazel build //java/arcs/core/data/testdata:WriterReaderExample_GeneratedPlan`
- It will produce a KT file. Copy + paste the contents to `src/tools/tests/goldens/WriterReaderExample.kt`
- Test to make sure this passes: `tools/sigh test -g recipe2plan`.

Reply:
Oh, yes, thanks!